presentation/profile/src/main/java/com/giraffe/profile/navigation/ProfileApiImp.kt

```diff
@@ -1,30 +1,46 @@
 package com.giraffe.profile.navigation
 
-import androidx.compose.foundation.layout.Box
-import androidx.compose.foundation.layout.fillMaxSize
 import androidx.compose.runtime.Composable
-import androidx.compose.ui.Alignment
-import androidx.compose.ui.Modifier
-import androidx.compose.ui.res.stringResource
-import com.giraffe.designsystem.composable.custom.Text
+import androidx.navigation.NavHostController
+import androidx.navigation.compose.rememberNavController
+import com.giraffe.details.DetailsApi
+import com.giraffe.explore.ExploreApi
 import com.giraffe.profile.ProfileApi
-import com.giraffe.profile.R
+import com.giraffe.profile.screens.collections.mycollections.MyCollectionsRoute
+import com.giraffe.profile.screens.settings.SettingsScreenRoute
 import javax.inject.Inject
 
-class ProfileApiImp @Inject constructor(): ProfileApi {
+class ProfileApiImp @Inject constructor(
+    private val detailsApi: DetailsApi,
+    private val exploreApi: ExploreApi,
+) : ProfileApi {
+
+    @Composable
+    override fun MyCollectionsContainer(
+        onShowBottomBarChange: (Boolean) -> Unit
+    ) {
+        val navController: NavHostController = rememberNavController()
+        ProfileNavGraph(
+            navController = navController,
+            startDestinationRoute = MyCollectionsRoute,
+            detailsApi = detailsApi,
+            exploreApi = exploreApi,
+            onShowBottomBarChange = onShowBottomBarChange
+        )
+    }
     @Composable
-    override fun ProfileContainer(onShowBottomBarChange: (Boolean) -> Unit) {
-        Box(
-            modifier = Modifier
-                .fillMaxSize()
-        ) {
-            Text(
-                modifier = Modifier
-                    .align(Alignment.Center),
-                text = stringResource(
-                    id = R.string.profile_screen
-                ),
-            )
-        }
+    override fun ProfileContainer(
```

Comment:
You duplicate the code of Container and change only route name
I Think is better to replace this duplicate by send parameter to Profile container to determine the start destination if setting Route Or CollectionRoute 
 Like this:
```suggestion
        override fun ProfileContainer(
        onShowBottomBarChange: (Boolean) -> Unit,
        toCollectionRoute : Boolean,
    ) {

        val navController: NavHostController = rememberNavController()
        val startDestination = if( toCollectionRoute) MyCollectionsRoute else SettingsScreenRoute

        ProfileNavGraph(
            navController = navController,
            startDestinationRoute = startDestination,
            detailsApi = detailsApi,
            exploreApi = exploreApi,
            onShowBottomBarChange = onShowBottomBarChange
        )
    }
```